### PR TITLE
Consider multi in add/remove/toggle select logic

### PR DESCRIPTION
**Pull request by sweco-sebhar**
Currently the add/remove/toggle logic will select all overlapping features regardless of the multi option. This means, for example, that with multi set to false all overlapping features will be selected with a single click if the shift key is pressed. With this change the multi option is considered similar to the "set" logic.

The background of this change has been discussed in #3402.

### CODE REVIEW

**Comment by sweco-sebhar** [comment]
@doronhorwitz, could you check if this PR makes things work as you expect?

**Comment by doronhorwitz** [comment]
This [jsfiddle](http://jsfiddle.net/doronhorwitz/022nnszd/3/) confirms that it works as expected, according to @tschaub's use cases in [#3402](https://github.com/openlayers/ol3/pull/3402#issuecomment-87701660). In the jsfiddle, the two leftmost squares are 'cats' and the rightmost square is a 'dog'. The 'dog' is also the top feature.
Toggling the ```multi``` between true and false and clicking around a lot also confirms that the behaviour has a good feeling.
In summary, I confirm things work as expected.

**Comment by elemoine** [comment]
@sweco-sebhar Do you think you could add unit-tests?

**Comment by sweco-sebhar** [comment]
@elemoine Tests added.

**Comment by elemoine** [comment]
I was expecting more tests to the `''filter features using the filter option'` section, but I can add them myself with a future PR.

**Comment by sweco-sebhar** [comment]
I've added tests to the 'filter features using the filter option' section but there is a failure I don't understand yet.

**Comment by sweco-sebhar** [comment]
Ok so the issue was that the non "set" logic that is used when pressing shift wasn't treating `multi: false` correctly when combined with feature filtering. I've fixed this so it should be the same as for "set" logic. Rebased and sqashed the branch.

**Comment by bjornharrtell** [comment]
Rebased and all seems well. Still hoping to get this in for 3.5.0, @elemoine?

**Comment by bjornharrtell** [comment]
Rebased on 3.5.0 master to keep this ready for review.

**Comment by ahocevar** [comment]
@elemoine Looks like you never finished reviewing this. Any chance you revisit?

**Comment by bjornharrtell** [comment]
Rebased but have failing tests so I'll need to put more time into this before it's ready for review again.

**Comment by elemoine** [comment]
@bjornharrtell, please ping me when this is ready to review again.

**Comment by bjornharrtell** [comment]
Finally got around to fix this PR. Ping @elemoine and previously involved parties @ahocevar, @tschaub since I suspect Éric has left the building.

**Comment by ahocevar** [comment]
This looks great now. Thanks @bjornharrtell for the extended effort.
